app/viewmodels/checkAnswers/PackagingSiteDetailsSummary.scala

```diff
@@ -0,0 +1,130 @@
+            content =
+              HtmlContent(addressFormatting(site)),
+            classes = "govuk-!-font-weight-regular govuk-!-width-two-thirds"
+          ),
+          actions = if (packagingSiteList.length > 1) {
+            Some(Actions("", Seq(
+              ActionItemViewModel("site.edit", routes.IndexController.onPageLoad().url) //TODO
+                .withVisuallyHiddenText(messages("packagingSiteDetails.hidden")),
+              ActionItemViewModel("site.remove", routes.IndexController.onPageLoad().url) //TODO
+                .withVisuallyHiddenText(messages("packagingSiteDetails.hidden"))
+            )))
+          } else {
+            Some(Actions("", Seq(
+              ActionItemViewModel("site.edit", routes.IndexController.onPageLoad().url) //TODO
+                .withVisuallyHiddenText(messages("packagingSiteDetails.edit.hidden"))
+            )))
+          }
+        )
+    }
+  }
+
+  private def addressFormatting(site: Site)(implicit messages: Messages): String = {
+    val commaFormattedSiteAddress = site.address.lines.map(line => { if (line.isEmpty) "" else line + ", " })
+
+    val addressNoTradingName = {
+      s"""${HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))}${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val addressWithTradingName = {
+      s"""${HtmlFormat.escape(site.tradingName.get)}<br>${HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))}
+      ${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val separatePostCodeAddressNoTradingName = {
+      s"""${HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))}<br>${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val separatePostCodeAddressWithTradingName = {
+      s"""${HtmlFormat.escape(site.tradingName.get)}<br>${
+        HtmlFormat.escape(commaFormattedSiteAddress.mkString(""))
+      }<br>${HtmlFormat.escape(site.address.postCode)}""".stripMargin
+    }
+
+    val addressFormat = determineAddressFormat(site)
+
+    addressFormat match {
+      case "separatePostCodeAddressNoTradingName" => separatePostCodeAddressNoTradingName
+      case "addressNoTradingName" => addressNoTradingName
+      case "addressWithTradingName" => addressWithTradingName
+      case "separatePostCodeAddressWithTradingName" => separatePostCodeAddressWithTradingName
+    }
+  }
+
+  private def determineAddressFormat(site: Site) = {
+    val addressLength = site.address.lines.toString().length
+    if (site.tradingName.get.isEmpty) {
+      if ((addressLength > 41 && addressLength < 47) || (addressLength > 97 && addressLength < 104)) {
+        "separatePostCodeAddressNoTradingName"
+      } else {
+        "addressNoTradingName"
+      }
+    } else {
+      if ((addressLength < 42 || addressLength > 46) && (addressLength < 98 || addressLength > 105)) {
+        "addressWithTradingName"
+      } else {
+        "separatePostCodeAddressWithTradingName"
+      }
```

Comment:
couldnt we just put each thing on a new line? is there a need for all of this logic